src/core/internal/array/construction.d

```diff
@@ -51,10 +52,17 @@ Tarr _d_arrayctor(Tarr : T[], T)(return scope Tarr to, scope Tarr from) @trusted
     {
         for (i = 0; i < to.length; i++)
         {
-            auto elem = cast(Unqual!T*)&to[i];
-            // Copy construction is defined as bit copy followed by postblit.
-            memcpy(elem, &from[i], element_size);
-            postblitRecurse(*elem);
+            static if (__traits(hasCopyConstructor, T))
+            {
+                emplace(&to[i], from[i]);
```

Comment:
Why use emplace and not just call the copy constructor?

Reply:
> Why use emplace and not just call the copy constructor?

I wasn't sure if `to[i] = T(from[i])` would create the object in-place at `to[i]` or it would create a temporary and a copy

> Isn't this eliding the call to the cpctor or is emplace taking care of this under the hood?

What do you mean? The cpctor gets called. The unittest checks for this.

Reply:
Ok, I was not sure what emplace does.

Reply:
Changed to call `__ctor`. This should probably be faster than calling `emplace`